src/shared/localShadowRepo.ts

```diff
@@ -44,27 +45,25 @@ interface CommitRequest {
 }
 
 export class ShadowRepo {
-  private static instance: ShadowRepo;
-
   public gitDir: string;
   public projectPath: string;
 
   private packageDirs!: NamedPackageDir[];
   private status!: StatusRow[];
   private logger!: Logger;
+  private gitIgnoreLocations: string[] = [];
 
   private constructor(options: ShadowRepoOptions) {
     this.gitDir = getGitDir(options.orgId, options.projectPath);
     this.projectPath = options.projectPath;
     this.packageDirs = options.packageDirs;
   }
 
+  // this is NOT a singleton but it was at one point, so its public method still sounds like it would be
   public static async getInstance(options: ShadowRepoOptions): Promise<ShadowRepo> {
```

Comment:
have you looked at current uses of this to make sure it isn't creating extra instances now that it's not a singelton?

Reply:
It's not exported, so if someone besides `SourceTracking.ts` is using it, that's their own fault.

SourceTracking.ts checks that it only inits/loads a ShadowRepo once via `ensureLocalSourceTracking`.

But I have a better idea...hang on.